### ISSUE TITLE
Internal HTML notelinks does not work in the mobile app

### ISSUE THREAD

**Issue by JackGruber** [laurent22/joplin#6515]
When an HTML link is used instead of the markdown notation to refer to another note, the link does not work in the mobile app.

## Environment

Joplin version: 2.7.2
Platform: Android 
OS specifics: 10

## Steps to reproduce

1. Create a note with the following content (Replace the ID by a reale noteid)
```
Works: [Test](:/cfbca1500bad4524a623092c8f598a67)
Works not: <a href=":/cfbca1500bad4524a623092c8f598a67">Test</a>
```
3. Test both links on the desktop app, both works
4. Test both links on the mobile app, only the first one works

## Describe what you expected to happen

Internal links with HTML notation should work

**Comment by NicoleSigaud** [comment]
Hello, I tried boths link on desktop app. The image opens in a third-party software (in my case, IrfanView) on both types of link.
![Sans titre](https://user-images.githubusercontent.com/6151758/169298008-4abaa1bb-f40f-48b5-a78c-c0427eb72691.png)
As for the Android 10 device, the same happened. No image shown, but the first link worked: the image opened alone, but not inside Joplin.
![Screenshot_2022-05-19-09-55-47-671](https://user-images.githubusercontent.com/6151758/169298902-5af5af84-406f-44ac-84be-08cd7c97eee5.jpeg)

**Comment by NicoleSigaud** [comment]
What I got so far for a link as image is nesting, which worked fine either on Windows and Android:
`[![](:/imagepath)](:/linkpath)`
As for the dimensions of the image, I found that
` [<img src=':/imagepath' style="width:25%; height:25%" />](:/linkpath)` -- works both on Android and Windows
The only thing is that the Joplin blue logo appears before the main image, indicating a link.

Now for NoteOverview, the last thing I tried was 
` text: "[{{image}}](:/{{id}})"` and that solved my problem

**Comment by Daeraxa** [comment]
@NicoleSigaud Your comments are about image links, the issue detailed here is about links to notes.

**Comment by NicoleSigaud** [comment]
Yes, I know, but this topic came exactly through a question I posted at https://discourse.joplinapp.org/t/note-overview/13962/265

**Comment by Tolu-Mals** [comment]
I was able to reproduce this on an android emulator and physical android device.

**Comment by NicoleSigaud** [comment]
@Tolu-Mals I'm glad you did! SO, as for the Joplin small logo before the links, it's there for good, isn't it?

**Comment by Tolu-Mals** [comment]
> @NicoleSigaud Yes

**Comment by laurent22** [comment]
@Tolu-Mals, did you look into this issue? What happens when the link is clicked - is there any error or message in the console? If not, you'll probably need to find where the click event is caught and how it is currently processed.

**Comment by Tolu-Mals** [comment]
Yes I looked into the issue. There's only a console warning "Possible unhandled promise rejection". 

I can see that the event is being processed inside the useOnMessage hook. I'll take a look at it.

I also plan to log out html that's been used in the Webview to see what I can find.

**Comment by Tolu-Mals** [comment]
So I logged the html generated in useSource to the console.

Output when using HTML Links
```
<p><a href=":/60ee5b0b0835484a887bb76e5bf9d109" class="jop-noMdConv">Note 1</a></p>
```

Output when using MD Notation Links
```
<p><a data-from-md data-resource-id='60ee5b0b0835484a887bb76e5bf9d109' href='#' ontouchstart='if (typeof(t) !== "undefined" && !!t) { clearTimeout(t); t = null; } else { t = setTimeout(() => { t = null; window.joplinPostMessage_("longclick:60ee5b0b0835484a887bb76e5bf9d109"); }, 500); }' ontouchend='if (!!t) {clearTimeout(t); t=null; window.joplinPostMessage_("joplin://60ee5b0b0835484a887bb76e5bf9d109");}' ontouchcancel='if (!!t) {clearTimeout(t); t=null; ontouchmove="if (!!t) {clearTimeout(t); t=null;'><span class="resource-icon fa-joplin"></span>First Note</a></p>
```

I'm still looking into why this happens and how to fix it.

**Comment by Tolu-Mals** [comment]
I discovered that the `link_open` rule doesn't work for html notation links. The `link_open` rule uses the `link_replacement` function to convert links into joplin file links, but this only works for markdown notation links.

I discovered this by adding `console.log()` inside the `link_open` rule. I get the log when a regular markdown link is present in the note, but not when a html notation link is present.

I'll read the documentation on markdown-it plugins to see how to fix this.

But I don't understand why the desktop app is able to open html notation joplin file links since it uses the same renderer the mobile app does. Does the desktop app handle links in a different way?
